//(C) Copyright [2020] Hewlett Packard Enterprise Development LP
//
//Licensed under the Apache License, Version 2.0 (the "License"); you may
//not use this file except in compliance with the License. You may obtain
//a copy of the License at
//
//    http:#www.apache.org/licenses/LICENSE-2.0
//
//Unless required by applicable law or agreed to in writing, software
//distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
//WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
//License for the specific language governing permissions and limitations
// under the License

syntax = "proto3";

service Authorization {
    rpc IsAuthorized(AuthRequest) returns (AuthResponse){}
}

message AuthRequest{
    string sessionToken = 1;
    repeated string privileges = 2;
    repeated string oemprivileges = 3;
}

message AuthResponse{
    int32 statusCode = 1;
    string statusMessage = 2;
}